apollo-router/src/context.rs

```diff
@@ -160,6 +160,21 @@ impl Context {
         result.map_err(|e| e.into())
     }
 
+    /// Upsert a JSON value in the context using the provided key and resolving
+    /// function.
+    ///
+    /// The resolving function must yield a value to be used in the context. It
+    /// is provided with the current value to use in evaluating which value to
+    /// yield.
+    pub fn upsert_json_value<K>(&self, key: K, upsert: impl Fn(Value) -> Value)
```

Comment:
Maybe just call this `upsert_value` as it could be any serde variant.
This also needs to be surfaced in a huddle. Maybe for now we just make this `pub(crate)`?

Reply:
No, this is specifically working with `serde_json_bytes::Value` directly. This skips the fallible serialize/deserialize parts of  `Context::upsert` that make is return a `Result`. This mirrors the way we have both `get_json_value` and `get`.

I’ll switch to `pub(crate)` for now.